File: client/src/services/invoiceService.js
import api from "../api/api";

export const saveInvoice = (invoice) =>
  api.post("/invoices", invoice);

export const getInvoices = () =>
  api.get("/invoices");

export const getInvoiceById = (id) =>
  api.get(`/invoices/${id}`);

export const updateInvoice = (id, invoice) =>
  api.put(`/invoices/${id}`, invoice);

export const deleteInvoice = (id) =>
  api.delete(`/invoices/${id}`);

// ✅ Mark Invoice as Paid
export const markInvoiceAsPaid = (id) =>
  api.put(`/invoices/${id}/paid`);

// ✅ Send Invoice Email
export const sendInvoiceEmail = (formData) =>
  api.post("/invoices/send-email", formData, {
    headers: {
      "Content-Type": "multipart/form-data",
    },
  });

File: server/src/main/java/server/service/InvoiceService.java
package server.service;

import org.springframework.data.domain.Page;
import server.entity.Invoice;

import java.util.List;
import java.util.Optional;

public interface InvoiceService {

    Invoice saveInvoice(Invoice invoice);

    List<Invoice> getAllInvoices();

    Optional<Invoice> getInvoiceById(String id);

    void deleteInvoice(String id);

    Invoice updateInvoice(String id, Invoice invoice);

    List<Invoice> searchInvoices(String invoiceNumber);

    Page<Invoice> getInvoices(int page, int size, String sortBy);

    // Mark invoice as Paid
    Invoice markAsPaid(String id);

}